### PR TITLE
[Feat] Filter Products

### PR DESCRIPTION
**Pull request by rushikesh3299**
Filter by price, category & ratings

Animation

### CODE REVIEW

File: src/components/products/Products.js [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+  };
+
+  const aboveRateProducts = (showRating, productList) => {
+    return [...productList].filter((item) => item.ratings >= showRating);
+  };
+
+  const sortedProducts = productState.sortBy
+    ? sortProducts(productState.sortBy, initialProductList)
+    : [...initialProductList];
+
```
Comment (mariyasada) [inline src/components/products/Products.js:27]: you can shift all these functions into a separate file

File: src/services/initial-products.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+
+export const initialProducts = () => {
+  const [initialProductList, setInitialProductList] = useState([]);
+
+  useEffect(async () => {
+    const data = await axios.get("/api/products");
+    setInitialProductList(() => data.data.products);
+  }, []);
```
Comment (mariyasada) [inline src/services/initial-products.js:10]: liked that you have created a separate file for products. you can destructure data here

**Review by mariyasada — APPROVED** [review]
This code looks fine to me. good to merge

File: src/components/filter/FilterMenu.js [hunk 3]
```diff
@@ -26,41 +50,69 @@ export const FilterMenu = () => {
...
+              type="checkbox"
+              name=""
+              id=""
+              checked={productState.showCategory.includes("Rasayana")}
+              onChange={() =>
+                productDispatch({ type: "CATEGORIZE", payload: "Rasayana" })
+              }
+            />
             Rasayana
           </label>
```
Comment (sushil-mod) [inline src/components/filter/FilterMenu.js:63]: this way is correct , just a suggestion but you can map over category array having all those categories that will remove the repetation of whole input html and their respective classes .

File: src/components/rating/Rating.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  for (let i = 0; i < filledStar; i++) {
+    starArray.push(<i className="fas fa-star"></i>);
+  }
+  for (let i = 0; i < halfStar; i++) {
+    starArray.push(<i className="fas fa-star-half-alt"></i>);
+  }
+  for (let i = 0; i < emptyStar; i++) {
+    starArray.push(<i className="far fa-star"></i>);
+  }
```
Comment (sushil-mod) [inline src/components/rating/Rating.js:15]: i liked how you implemented rating component

**Review by sushil-mod — APPROVED** [review]
just a suggestion . overall code is good . In preview show any links not working i am not able to go on product page ( check it once). Rest is Good UI

**Pull request merged**